hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFilePrettyPrinter.java

```diff
@@ -323,15 +323,20 @@ public int processFile(Path file, boolean checkRootDir) throws IOException {
       HFileScanner scanner = reader.getScanner(false, false, false);
       fileStats = new KeyValueStatsCollector();
       boolean shouldScanKeysValues = false;
+      boolean needSeekToBegin = true;
       if (this.isSeekToRow) {
         // seek to the first kv on this row
-        shouldScanKeysValues =
-          (scanner.seekTo(PrivateCellUtil.createFirstOnRow(this.row)) != -1);
-      } else {
+        if (scanner.seekTo(PrivateCellUtil.createFirstOnRow(this.row)) != -1) {
+          needSeekToBegin = false;
+          shouldScanKeysValues = true;
+        }
+      }
+      if (needSeekToBegin) {
```

Comment:
Isn't this `scanner.seekTo(PrivateCellUtil.createFirstOnRow(this.row))` always going to return -1, since it's fabricating a KV with no CF? If so, I guess your logic could be simplified.

Reply:
Gotcha, thanks for clarifying!